[ZEPPELIN-6358] Add E2E tests about /#/notebook/:noteid for New UI

voidmatcha opened the pull request:
### What is this PR for?
Addition and improvement of notebook-related E2E tests for New UI

---

/#/notebook/:noteId – View or edit a specific notebook
/#/notebook/:noteId/revision/:revisionId – View a specific revision of a notebook
/#/notebook/:noteId/paragraph/:paragraphId – Notebook paragraph presentation mode

PAGES.WORKSPACE.NOTEBOOK
→ src/app/pages/workspace/notebook/notebook.component

PAGES.WORKSPACE.NOTEBOOK_ACTION_BAR
→ src/app/pages/workspace/notebook/action-bar/action-bar.component

PAGES.WORKSPACE.NOTEBOOK_PARAGRAPH
→ src/app/pages/workspace/notebook/paragraph/paragraph.component

PAGES.WORKSPACE.NOTEBOOK_SIDEBAR
→ src/app/pages/workspace/notebook/sidebar/sidebar.component

PAGES.WORKSPACE.PUBLISHED_PARAGRAPH
→ src/app/pages/workspace/published/paragraph/paragraph.component

### What type of PR is it?
Improvement

### Todos

### What is the Jira issue?
ZEPPELIN-6358

### How should this be tested?

### Screenshots (if appropriate)

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

voidmatcha committed:
Add notebook related tests

voidmatcha committed:
combine tests

voidmatcha committed:
add shortcut tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
remove unsued function and apply review

voidmatcha committed:
add global teardown's cleanup test notebooks

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-keyboard-page.ts:
```diff
@@ -0,0 +1,1056 @@
+
+  async navigateToNotebook(noteId: string): Promise<void> {
+    if (!noteId) {
+      console.error('noteId is undefined or null. Cannot navigate to notebook.');
```
It seems better to throw an error right away when `noteId` is falsy, since that implies an earlier-step issue, so we don't end up silencing it. WDYT?

voidmatcha committed:
cleanupTestNotebooks only for local, add waitForSelector

voidmatcha committed:
enhance tests

tbonelee commented:
Could you rebase this onto master branch?

tbonelee reviewed (changes requested):
Could you take a quick pass on two areas?
1. There are places where try/catch accepts both error and non-error paths. Could we make these more explicit, either intentionally silencing with a clear rationale or failing, so we reduce false positives?
2. For browser-specific branches (e.g., notebook-keyboard-page.ts → executePlatformShortcut), could we extract that logic so the main flow reads sequentially and the differences are isolated?

voidmatcha committed:
apply lint:fix

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
verify icon updates by asserting svg data-icon attribute

voidmatcha committed:
make verifyTitleEditingFunctionality assert actual editing behavior

voidmatcha committed:
lint fix + throw error

voidmatcha committed:
extract browser-specific logic in executePlatformShortcut for clarity

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
add missing annotation

voidmatcha committed:
add additional tests

voidmatcha committed:
fix broken tests

tbonelee commented on zeppelin-web-angular/e2e/models/note-toc-page.ts:
```diff
@@ -0,0 +1,119 @@
+
+  constructor(page: Page) {
+    super(page);
+    this.tocToggleButton = page.locator('.sidebar-button').first();
```
How about adding more specific label(e.g., `aria-label`) to the element and using that to select the element?

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,218 @@
+    if (hasResult) {
+      await expect(this.paragraphPage.resultDisplay).toBeVisible();
+    }
+  }
```
It seems like always-passing tests.

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,218 @@
+          console.log('Title editor is a custom editable component');
+        } else {
+          console.log('Title editor may not be editable in current state');
+        }
```
I think this catch caluse may lead to false postive.

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-sidebar-page.ts:
```diff
@@ -0,0 +1,423 @@
+      } catch {
+        // Neither TOC nor file tree visible
+      }
+    }
```
Could we use `waitFor()` method in `Locator` object, instead of wrapping `expect()` method?

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/published/published-paragraph.spec.ts:
```diff
@@ -87,55 +94,139 @@ test.describe('Published Paragraph', () => {
+      const isResultVisible = await page.locator('zeppelin-notebook-paragraph-result').isVisible();
+      if (isResultVisible) {
+        await expect(page.locator('zeppelin-notebook-paragraph-result')).toBeVisible();
+      }
```
Always-passing test.

tbonelee reviewed (changes requested):
Thank you for making the changes.

https://github.com/tbonelee/zeppelin/commits/PR_5101-refactor/
This branch is where I tried removing only unused methods and variables in bulk.
It would be great if you could take a look and apply the safe-to-delete items.

And I’d like to share a few personal thoughts.

Since a single method handles multiple branches, the test scenarios and logic seem to become quite complex.
Similarly, within one method, there are cases where it’s supposed to fail in some situations and succeed in others, which leads to more try/catch blocks inside test methods and, in some cases, unintentionally suppressing failures.

Overall, I’d like to suggest moving in the following direction:
- If tests need to handle branching, consider splitting the method itself.
- Reduce if statements by explicitly identifying/controlling the situation when possible.
- Be cautious about wrapping failures in try/catch within test methods.

I’m not suggesting rewriting everything; rather, fixing what can be improved in this PR and proposing a direction for future PRs.

If you have any other ideas, feel free to share them!

voidmatcha committed:
remove unused part by tbonelee
Squashed commit of the following:

commit c230d3c7c40acdc23dc2ef9d401cef03e896e093
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 23:16:41 2025 +0900

    Remove unused `verifyAllSidebarFunctionality` method from `NotebookSidebarPage` model in E2E tests

commit 0a0b09f114450ba49b718b398108219d259b6a79
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 23:06:17 2025 +0900

    Remove unused methods and variables from `NotebookSidebarPage` model in E2E tests

commit a2afa6f8cc39713250fb9ea7dc208c32f5461402
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 23:03:58 2025 +0900

    Remove unused methods from `NotebookParagraphPage` model in E2E tests

commit 5ef2f846420b2c3c2ab3f97ff7b77538b201dbdf
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:51:25 2025 +0900

    Remove unused methods and variables from `NotebookParagraphPage` model in E2E tests

commit cbd0c87328444322b7d3b64b24b4d9d7f7a98345
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:48:54 2025 +0900

    Remove unused methods, variables, and constructor logic from `NotebookPageUtil` and `NotebookPage` models in E2E tests

commit 7b4c09dbf6c10a62b3ab59f69bbac8ac826d75c9
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:44:10 2025 +0900

    Remove unused methods and `sidebar` locator from `NotebookPage` model in E2E tests

commit 1e2459bb57ad860a5688cfb8f88a4acb49e108e4
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:42:24 2025 +0900

    Remove unused methods from `NotebookKeyboardPage` model in E2E tests

commit 852831136904cef602157555e5affde3bdfd2367
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:40:19 2025 +0900

    Remove unused methods from `NotebookKeyboardPage` model in E2E tests

commit 238304d0071489d11dc20483c36370e3143efe32
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:18:27 2025 +0900

    Remove unused `verifyCommitWorkflow` method from `NotebookActionBarPage` model in E2E tests

commit 744584919c2a882a83438241535fd740bd0a6bdd
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:17:40 2025 +0900

    Remove unused methods and variables from `NotebookActionBarPage` model in E2E tests

commit 4f44e3d80d6526b5d05eb806d150fd1d835ba292
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:09:37 2025 +0900

    Remove unused methods and redundant constructor argument from note TOC page util in E2E tests

commit d7a991225380aff70eabc7ff0da36547e8aaea42
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:07:14 2025 +0900

    Remove unnecessary overriding parent properties

commit f4d8d256e6cbbfb16308ce0ffed57dba14d16b73
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 22:06:30 2025 +0900

    Remove unused methods, variables, and constructor logic from note TOC page model in E2E tests

commit 379e497871dedb5dde4d977d12ce24631ebb7249
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 21:47:24 2025 +0900

    Remove unused methods and `navigate` from note rename page model

commit b28cf71cdaf3c11214bf1ec4a0375e5e8acb7f94
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 21:35:20 2025 +0900

    Remove redundant variables and empty catch block in folder rename page model

commit 44a0c7b014e3ed9da53ffcd81a5dfef2ac29d75d
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 21:31:13 2025 +0900

    Remove unused methods and navigate function from folder rename page model

commit 2a67d5cc98c1a1041de3a376a46aaeb6ce26ea5f
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 21:30:06 2025 +0900

    Remove unused `HomePage` import and associated variable from folder rename test

commit 578844b652c368d1ec1b2f4248747ca9e8f696ce
Author: ChanHo Lee <chanholee@apache.org>
Date:   Thu Nov 6 21:28:55 2025 +0900

    Make folderName parameter mandatory

voidmatcha committed:
fix sidebar-functionality.spec related tests

voidmatcha committed:
add sidebar aria-label

voidmatcha committed:
apply published-paragraph related tests

voidmatcha committed:
apply review about notebook-paragraph and sidebar

voidmatcha committed:
apply review rest of tests

voidmatcha committed:
fix action bar relate tests

voidmatcha committed:
fix published paragraph test

voidmatcha committed:
fix notebook paragraph related tests

voidmatcha committed:
fix tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
add tear up and tear down related about ZEPPELIN_NOTEBOOK_DIR

voidmatcha committed:
fix borken test

voidmatcha committed:
fix test pom about keyboard and sidebar

voidmatcha committed:
remove try catch

voidmatcha committed:
remove conditional test

voidmatcha committed:
remove try catch

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix remove condition of test

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix broken test

voidmatcha committed:
add ci settings for python

voidmatcha committed:
remove conditional skip tests

voidmatcha committed:
fix interpreter not found error

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix issue where Control+Alt+F dialog does not appear

voidmatcha committed:
fix broken tests

voidmatcha committed:
conver to use createTestNotebook deleteTestNotebook from utils

voidmatcha committed:
fix broken tests

voidmatcha reviewed (commented):
First, I want to apologize for putting so many changes into a single PR. There were some issues caused by relying on Playwright MCP and LLM. Currently, all tests seem to be passing except for four that are still failing.

---

The `try/catch` blocks have been removed. For tests with conditional logic, I removed as much as possible where it was safe to do so.

---

Regarding `cleanupTestNotebooks`, the idea of using `ZEPPELIN_NOTEBOOK_DIR` was applied only in CI because applying it in local tests would require restarting the server. f59c548ba1e11419ca90991f1f80f556b988c0e5

---

In CI, running paragraphs with Python caused an “Interpreter not found” error, so I added the necessary configuration to address this. be6aebd4f09a89fffc4592cfdafcd555c2bcd568

---

<img width="240" height="87" alt="Image" src="https://github.com/user-attachments/assets/16599001-8616-490e-984c-d697328b3b48" />

For folder renames, above image type of error was occurring, so this was also handled appropriately in CI. a25c3f947b91bfac498677a055b452ec8c1e7ae7

---

There was an issue where shortcut keys were not connected about notebook searches, so this was fixed. 955b4d69d87c742bbd1b4402e4406c721f1de3b8

---

The four tests that are still failing are as follows. They mostly seem to fail because the shortcut inputs are not being correctly registered—possibly due to Monaco Editor intercepting key inputs.(ArrowUp, ArrowDown, CutLine, SearchInsideCode, ...) The exact cause is still unclear, so a separate issue has been filed for this:

* `ParagraphActions.RunAbove`: Control+Shift+ArrowUp
* `ParagraphActions.RunBelow`: Control+Shift+ArrowDown
* `ParagraphActions.CutLine`: Control+K
* `ParagraphActions.FindInCode`: Control+Alt+F

ZEPPELIN-6379

```ts
// notebook-paragraph-keyboard-event-handler.ts
...
const MonacoHandledParagraphActions = [
  ParagraphActions.MoveCursorUp,
  ParagraphActions.MoveCursorDown,
  ParagraphActions.SwitchEditor,
  ParagraphActions.CutLine,
  ParagraphActions.PasteLine,
  ParagraphActions.SearchInsideCode
] as const;
...
```
This is consistent with the code section.

I’m not sure why, but it works in WebKit. It doesn’t work in the other browsers. <- on CI

Since it passes on all browsers locally, it might be an issue with how CI's OS(Ubuntu) detects keyboard shortcuts.

voidmatcha committed:
disable the button in the rename modal when the input field is empty

voidmatcha committed:
fix broken tests

voidmatcha committed:
test

tbonelee commented on zeppelin-web-angular/e2e/models/folder-rename-page.util.ts:
```diff
@@ -0,0 +1,231 @@
+
+    const renameButton = folderNode.locator('a[nz-tooltip][nztooltiptitle="Rename folder"]');
+    await expect(renameButton).toBeVisible();
+  }
```
Could you please check these duplicates again? I think it is sufficient to check the individual buttons('Create new note', 'Rename', 'Delete').

tbonelee commented on zeppelin-web-angular/e2e/models/note-rename-page.ts:
```diff
@@ -0,0 +1,58 @@
+    super(page);
+    // Note title in elastic input component
+    this.noteTitle = page.locator('.elastic p');
+    this.noteTitleInput = page.locator('.elastic input');
```
```suggestion
    this.noteTitle = page.getByRole('heading').locator('p');
    this.noteTitleInput = page.getByRole('heading').locator('input');
```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-action-bar-page.ts:
```diff
@@ -0,0 +1,130 @@
+
+  async confirmCommit(): Promise<void> {
+    await this.commitConfirmButton.click();
+  }
```
```suggestion
  }

```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-keyboard-page.ts:
```diff
@@ -0,0 +1,674 @@
+      return; // Exit early if the page is already closed
+    }
+
+    const browserName = this.page.context().browser()?.browserType().name();
```
```suggestion
```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-page.util.ts:
```diff
@@ -0,0 +1,87 @@
+
+  async verifyExtensionAreaWhenActivated(): Promise<void> {
+    await this.verifyExtensionAreaIfVisible();
+  }
```
Why do these methods only call another method?
Are they intended as simple aliases, or can we use the original methods directly?

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.ts:
```diff
@@ -0,0 +1,95 @@
+
+  async doubleClickToEdit(): Promise<void> {
+    await this.paragraphContainer.dblclick();
+  }
```
```suggestion
  }

```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.ts:
```diff
@@ -0,0 +1,95 @@
+  }
+  async runParagraph(): Promise<void> {
+    await this.runButton.click();
+  }
```
```suggestion
  }

```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,182 @@
+  async verifyParagraphControlInterface(): Promise<void> {
+    await expect(this.paragraphPage.controlPanel).toBeVisible();
+    await this.paragraphPage.runButton.isVisible();
+    await expect(this.paragraphPage.runButton).toBeVisible();
```
```suggestion
    await expect(this.paragraphPage.runButton).toBeVisible();
```

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,182 @@
+  async verifyCodeEditorFunctionality(): Promise<void> {
+    await this.paragraphPage.isCodeEditorVisible();
+    await expect(this.paragraphPage.codeEditor).toBeVisible();
+  }
```
We might change the term `Functionality` to `Visibility` and test the evaluation of the `await this.paragraphPage.isCodeEditorVisibile()`.

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,182 @@
+  async verifyResultDisplaySystem(): Promise<void> {
+    await this.paragraphPage.hasResult();
+    await expect(this.paragraphPage.resultDisplay).toBeVisible();
+  }
```
Not testing the result of the expression `await this paragraphPage.hasResult()` has no effect to the method result.

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-paragraph-page.util.ts:
```diff
@@ -0,0 +1,182 @@
+        '⚠️ Dynamic forms verification skipped: Interpreter not available or error occurred. This test requires proper interpreter configuration.'
+      );
+      return;
+    }
```
Shouldn't we fail fast in this case?

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+      await keyboardPage.waitForParagraphExecution(0);
+      const hasResult = await keyboardPage.hasParagraphResult(0);
+      expect(hasResult).toBe(true);
+    });
```
Maybe we could merge these into a single method.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+
+      // Then: Paragraph count should decrease
+      const finalCount = await keyboardPage.getParagraphCount();
+      expect(finalCount).toEqual(1);
```
Should `finalCount` be expected to `currentCount - 1`?

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+      // Then: Paragraph order should change (first becomes second)
+      await keyboardPage.page.waitForTimeout(1000);
+      const paragraphCount = await keyboardPage.getParagraphCount();
+      expect(paragraphCount).toBe(2);
```
We should test paragraph positions.

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-keyboard-page.ts:
```diff
@@ -0,0 +1,674 @@
+    return await runningIndicator.isVisible();
+  }
+
+  async hasParagraphResult(paragraphIndex: number = 0): Promise<boolean> {
```
The method name might be misread as checking only the existence of the paragraph result element.
Since the implementation actually checks whether the result’s status is in a determinable state, maybe a clearer name would be something like `isParagraphResultSettled`.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+
+      // Then: Paragraph width should change
+      const finalWidth = await keyboardPage.getParagraphWidth(0);
+      expect(finalWidth).not.toBe(initialWidth);
```
The same as the above.

```suggestion
      expect(finalWidth).toBeGreaterThan(initialWidth);
```

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+      // Then: Content should be pasted back
+      await keyboardPage.page.waitForTimeout(500);
+      const finalContent = await keyboardPage.getCodeEditorContent();
+      expect(finalContent.length).toBeGreaterThan(0);
```
Could we also assert that the content actually changes?
For example, after `Cut Line` we check that the editor content is reduced (something disappears), then clear the input and verify it’s empty, and finally after `Paste Line` assert that the original content is restored.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+      // Then: System should remain stable
+      const codeEditorComponent = keyboardPage.page.locator('zeppelin-notebook-paragraph-code-editor').first();
+      await expect(codeEditorComponent).toBeVisible();
+    });
```
I may be missing the intention here, but what specific behavior are we trying to verify in this test?
As it is now, the Then step seems like it would always succeed, so I might be misunderstanding what failure case this is meant to cover.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts:
```diff
@@ -0,0 +1,867 @@
+      // Then: Code should contain interpreter directive
+      const content = await keyboardPage.getCodeEditorContent();
+      expect(content).toContain('%python');
+    });
```
Just to confirm, is this test mainly checking that keyboard input is correctly delivered to the editor?
The title suggests it’s about interpreter selector shortcuts, but the current assertion looks like it’s only verifying that typing works.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/main/notebook-container.spec.ts:
```diff
@@ -0,0 +1,88 @@
+test.describe('Notebook Container Component', () => {
+  addPageAnnotationBeforeEach(PAGES.WORKSPACE.NOTEBOOK);
+
+  let testUtil: PublishedParagraphTestUtil;
```
```suggestion
```

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/published/published-paragraph.spec.ts:
```diff
@@ -87,55 +96,135 @@ test.describe('Published Paragraph', () => {
+      // Verify we're actually in published mode by checking page structure
+      const isPublishedMode = await page.evaluate(() => document.querySelector('zeppelin-publish-paragraph') !== null);
+      expect(isPublishedMode).toBe(true);
+    });
```
I might be missing the intention, but what does "maintain paragraph context" mean in this test?
It would be helpful to clarify what behavior we expect here when the page is in published mode.

tbonelee commented on zeppelin-web-angular/e2e/tests/notebook/published/published-paragraph.spec.ts:
```diff
@@ -87,55 +96,135 @@ test.describe('Published Paragraph', () => {
+      const isDynamicFormsVisible = await page.locator('zeppelin-notebook-paragraph-dynamic-forms').isVisible();
+      if (isDynamicFormsVisible) {
+        await expect(page.locator('zeppelin-notebook-paragraph-dynamic-forms')).toBeVisible();
+      }
```
Just to clarify the intention here: if the dynamic forms are not visible, this check won’t fail the test.
Are we intentionally allowing that, or did we want to assert that dynamic forms must be visible in published mode?

tbonelee reviewed (changes requested):
Thank you again for continuing to work on this.
One thing I wanted to mention is that the PR has become quite large 😅
Would it be possible to apply this round of feedback and then split the changes into smaller PRs?

If the pieces are not fully independent, stacking them like a stacked diff would also be totally fine.

My personal suggestion would be to keep larger spec files separate and group only smaller spec files that belong to the same area.

Please feel free to share any thoughts or preferences you have.

tbonelee committed:
Apply minor suggestions from code review

tbonelee committed:
Apply minor suggestions from code review

tbonelee committed:
Remove console log

tbonelee committed:
Remove unused method

tbonelee committed:
Reformat blank lines

tbonelee commented on zeppelin-web-angular/e2e/models/notebook-action-bar-page.ts:
```diff
@@ -0,0 +1,132 @@
+  async confirmCommit(): Promise<void> {
+    await this.commitConfirmButton.click();
+  }
+
```
```suggestion
```

tbonelee committed:
Reformat blank lines

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix borken tests

voidmatcha committed:
remove r setup in frontend.yml playwright step

voidmatcha committed:
reflecting a simple review

voidmatcha committed:
fix broken tests

voidmatcha committed:
added a comment to the additionally assigned issue.

voidmatcha committed:
refactor error handling to fail fast instead of catching and continuing

voidmatcha committed:
change goto('/') to goto('/#/')

voidmatcha committed:
fix broken test

voidmatcha committed:
remove deleteNotebook func and step for afterEach

voidmatcha committed:
refactor teardown step about delete note

voidmatcha committed:
refactor tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
refactoring

voidmatcha committed:
fix broken tests

voidmatcha committed:
refactoring and fix broken tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
change playwright config ts to js

voidmatcha committed:
change to folder base note managing and cleanup

voidmatcha committed:
add notebook related methods in utils

voidmatcha committed:
add comment for globl-teardown

voidmatcha committed:
remove unnecessary code

voidmatcha committed:
utils convert to arrow func, separate regex to  constant

voidmatcha committed:
fix borken tests

voidmatcha committed:
reset unnecessary changes about package.json

voidmatcha committed:
add safe step for watiForNotebookLinks

voidmatcha committed:
fix broken tests

voidmatcha committed:
change e2eTestFolder selector on base-page

voidmatcha committed:
refactoring performLoginIfRequired for broken tests

voidmatcha committed:
fix broken tests

voidmatcha committed:
add expandfolder step in getFolderNode

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test about Control+Alt+F

voidmatcha committed:
fix broken test about rename folder

voidmatcha committed:
fix broken test

voidmatcha committed:
move constants location and make a new constant

voidmatcha committed:
add comments for ZEPPELIN_E2E_TEST_NOTEBOOK_DIR

voidmatcha committed:
simplize Setup Test Notebook Directory step in CI

voidmatcha committed:
separate constatns to constant file

voidmatcha committed:
rollback constants location for circular dependency issue

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix env config in yml

voidmatcha committed:
fix path condition in cleanupTestNotebooks

voidmatcha committed:
fix broken test

voidmatcha committed:
add waitForZeppelinReady step  in performLoginIfRequired

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix constants path

voidmatcha committed:
fix broken test

voidmatcha committed:
updated the folder rename test to run outside of E2E_TEST_FOLDER for stability

voidmatcha committed:
Ctrl+Y marked as skipped in CI - ZEPPELIN-6379

voidmatcha committed:
apply review

voidmatcha committed:
apply review

voidmatcha committed:
fix broken tests

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
fix broken test

voidmatcha committed:
apply review

voidmatcha committed:
apply review

voidmatcha committed:
remove unused code

voidmatcha committed:
refactoring tests

voidmatcha committed:
fix broken test

voidmatcha committed:
revert unproper change for selenium ci step

voidmatcha commented:
I had forgotten about this, but since the changes in this PR have already been split up and merged into master, I’ll close it.
